bal/p2p.py

```diff
@@ -156,24 +195,49 @@ def start(self):
-                    self.query(peer_addr, Message(MessageType.QUERY_LATEST_BLOCK, '', self.p2p_addr))
+                if peer_str not in self.peer_sockets:
+                    self.query(
+                        peer_addr,
+                        Message(
+                            MessageType.QUERY_LATEST_BLOCK,
+                            '',
+                            self.p2p_addr))
                     print('Added new peer {}'.format(peer_str))
 
                 if message.type == MessageType.RESPONSE_BLOCKCHAIN:
                     self.process_response_chain(message)
                 elif message.type == MessageType.QUERY_ALL:
-                    print('Dispatching all blocks to {}'.format(message.reply_addr[1]))
-                    self.query(peer_addr, Message(MessageType.RESPONSE_BLOCKCHAIN, self.blockchain().get_blockchain(), self.p2p_addr))
+                    print(
+                        'Dispatching all blocks to {}'.format(
+                            message.reply_addr[1]))
+                    self.query(
+                        peer_addr,
+                        Message(
+                            MessageType.RESPONSE_BLOCKCHAIN,
+                            self.blockchain().get_blockchain(),
+                            self.p2p_addr))
                 elif message.type == MessageType.QUERY_LATEST_BLOCK:
-                    print('Dispatching latest block to {}'.format(message.reply_addr[1]))
-                    self.query(peer_addr, Message(MessageType.RESPONSE_BLOCKCHAIN, [self.blockchain().get_latest_block()], self.p2p_addr))
+                    print(
+                        'Dispatching latest block to {}'.format(
+                            message.reply_addr[1]))
+                    self.query(
+                        peer_addr, Message(
+                            MessageType.RESPONSE_BLOCKCHAIN, [
+                                self.blockchain().get_latest_block()], self.p2p_addr))
                 elif message.type == MessageType.QUERY_TRANSACTION_POOL:
-                    self.query(peer_addr, Message(MessageType.RESPONSE_TRANSACTION_POOL, self.transaction_pool().get_transaction_pool(), self.p2p_addr))
+                    self.query(
+                        peer_addr,
+                        Message(
+                            MessageType.RESPONSE_TRANSACTION_POOL,
+                            self.transaction_pool().get_transaction_pool(),
+                            self.p2p_addr))
                 elif message.type == MessageType.RESPONSE_TRANSACTION_POOL:
                     received_transactions = message.data
                     if not received_transactions:
-                        print('invalid transaction received: %s', json.dumps(message.data))
+                        print(
+                            'invalid transaction received: %s',
+                            json.dumps(
+                                message.data))
```

Comment:
same pattern for splitting occurs until here. let's consider to use more compact lines. or at least find more suitable spot to split these lines.